Update xmllint download url.

nuclearsandwich opened the pull request:
There are two commits here, one which refactors the downloads to use a single url path set via chef attribute and a second to change that attribute to point to the ROS download server (via OSUOSL FTP endpoint for proper https support) in order to work around outdated root CA info in the chef version used in these builds. Updating Chef is another more complete solution which will take time. This is meant to get us running in the meantime.

nuclearsandwich committed:
Make xmllint download url an attribute.

This makes it easier to change the xmllint download url in configuration
rather than via code changes.

nuclearsandwich committed:
Use ROS download server for xmllint downloads.

As a workaround for some host SSL issues we have rehosted these binaries
on a host which does not use Let's Encrypt for HTTPS certificates.

clalancette commented on cookbooks/ros2_windows/recipes/xmllint.rb:
```diff
@@ -1,18 +1,19 @@
 seven_zip_archive 'libxml2' do
   path 'C:\\xmllint'
-  source 'https://www.zlatkovic.com/pub/libxml/64bit/libxml2-2.9.3-win32-x86_64.7z'
+  source xmllint_url + 'libxml2-2.9.3-win32-x86_64.7z'
```
This is a minor nit, but using `+` here seems a little fragile.  If someone forgets to add the `/` to the end of the download_url, this will produce an invalid URL.  Does ruby/chef have the equivalent of `'/'.join([a, b])` that Python has?

nuclearsandwich replied:
It does
```suggestion
  source [xmllint_url, 'libxml2-2.9.3-win32-x86_64.7z'].join('/')
```

and that would have the much less fatal effect of doubling the '/' if there is one present.
From a "keep it as obvious as possible" perspective I would probably opt for adding a leading slash to the file name and maintaining the concatenation rather than using a join. It has the additional, completely pointless, advantage of eliminating an additional array and string allocation (the array enclosing the url prefix and filename, and the literal `'/'`). 

```suggestion
  source xmllint_url + '/libxml2-2.9.3-win32-x86_64.7z'
```

clalancette reviewed (approved):
I left one nit to think about, but I'm going to approve anyway.  You can decide whether to address it or not.

nuclearsandwich committed:
Guard against a url parameter with no trailing slash.

This will result in a doubled slash in the url in some cases but that
should be benign to most web servers.

nuclearsandwich committed:
Revert "Use ROS download server for xmllint downloads."

This reverts commit a5f9583995f2cb106a9c0e82b2fc20f7ada74d34.

This was not a sufficient temporary workaround.
